bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -161,73 +161,50 @@ private void checkAssetsLoaded() throws AssetNotLoadedException {
     private synchronized void checkConnection() {
         logger.debug("checkConnection()");
 
-        // check connection to the hub
-        ThingStatusDetail thingStatus;
+        boolean retryApplicationKey = false;
+        boolean retryConnection = false;
+
         try {
             checkAssetsLoaded();
             getClip2Bridge().testConnectionState();
-            thingStatus = ThingStatusDetail.NONE;
-        } catch (HttpUnauthorizedException e) {
-            logger.debug("checkConnection() {}", e.getMessage(), e);
-            thingStatus = ThingStatusDetail.CONFIGURATION_ERROR;
+            updateSelf(); // go online
+        } catch (HttpUnauthorizedException unauthorizedException) {
+            logger.debug("checkConnection() {}", unauthorizedException.getMessage(), unauthorizedException);
+            if (applKeyRetriesRemaining > 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.api2.conf-error.press-pairing-button");
+                try {
+                    registerApplicationKey();
+                    retryApplicationKey = true;
```

Comment:
Surely retryApplicationKey should be false here as the registration will have succeeded. Or?

Reply:
You might be right, but I didn't want to change behavior in context of this PR. If we can get this PR merged (to avoid conflicts), would you do a follow-up PR for this?

Reply:
As I said in my prior post, this is quite critical functionality, so we should not merge hastily until it has been fully tested. Which I cannot do until I return next week. I was happy with the PR until your last commit, so if you want to merge quickly then I suggest to revert that last commit.

Reply:
Let's not rush, it can wait.

Reply:
Resolving comment since fixing application key logic is outside the scope of this PR.

Reply:
Ok. I checked it. The use of `retryApplicationKey` is correct, so no changes required.